rustc: 1.34.2 -> 1.35.0

edolstra opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

edolstra committed:
rustc: 1.34.2 -> 1.35.0

edolstra committed:
rustfmt: 1.2.1 -> 1.2.2

Mic92 commented:
`fd` builds and runs on aarch64.

Mic92 commented:
macOS seems to have a build problem:

```
error: install_name_tool: changing install names or rpaths can't be redone for: /nix/store/mxb4iiivxv30qvwi7z21xkzhz28175hc-rustc-bootstrap-1.34.2/lib/librustc_typeck-f577d18db8fe4627.dylib (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
```

LnL7 committed:
rustc: don't patch precompiled binaries on darwin

This doesn't work anymore since 1.35.0 because the load commands end up
being too large, but this is a precompiled binary anyway so it's not
really needed or more impure to use the expected system versions.

LnL7 commented on pkgs/development/compilers/rust/binaryBuild.nix:
```diff
@@ -85,7 +75,8 @@ rec {
     };
 
-    buildInputs = [ makeWrapper bash ] ++ stdenv.lib.optional stdenv.isDarwin Security;
+    buildInputs = [ makeWrapper bash ]
```
@Mic92 do you remember why you added bash here?

Mic92 replied:
@LnL7 also looks suspicious to me. It would only make a difference in a cross-compiled build. We can properly drop it.

Mic92 reviewed (approved):
If macOS builds, we can merge this!

LnL7 commented:
Yep, has anyboy tested firefox?

Mic92 commented:
I was able to build firefox.
